### ISSUE TITLE
TLS, DOCKER_HOST and authenticate server based on given CA

### ISSUE THREAD

**Issue by qk4l** [fabric8io/docker-maven-plugin#400]
Hello,

I'm trying to configure environment to use `docker-maven-plugin` with remote `DOCKER_HOST` and communicate with last via TLS.

As I read Docker provide [several modes](https://docs.docker.com/engine/security/https/#client-modes) to authenticate clients. Just because my `DOCKER_HOST` uses trusted certificate and I don't want to generate certificates for all pack of agents (builders), the authenticate server based on given CA is the best mode as I see.

So I've ran `DOCKER_HOST` with `tls`, `tlscert`, `tlskey` flags and put `ca.pem` file in `DOCKER_CERT_PATH` directory on agents. But plugin are trying to read `key.pem` and `cert.pem` files as it wants to authenticate with client certificate.

Could you please explain if it possible now to use plugin without client certificate by stay secure and checking `DOCKER_HOST` certificate?

Thank you.

**Comment by rhuss** [comment]
This is a good point. Currently only the secure mode with client cert **and** CA validation is supported, but it should be easy to support a CA validation only mode (as well as a client-cert only mode).

However in this mode you must ensure, that you daemon does not authenticate clients then (which effectively allows everybody  to use your docker daemon and fetch and run e.g. all images there).

When implementing this it is probably also a good idea to add a configuration option `tlsVerify` in addition to evaluating the environment variable `DOCKER_TLS_VERIFY` as it is now.

**tl;dr**  - Currently this is not possible but would be a good addition. PRs always welcome, which increases the likelihood to be included in one of the nexts releases ;-) (but we will add it anyways)
